src/Chainweb/BlockHeaderDB/RestAPI/Server.hs

```diff
@@ -313,11 +313,14 @@ headerStreamServer cdb = headerStreamHandler cdb
 
 headerStreamHandler :: forall cas. PayloadCas cas => CutDb cas -> Tagged Handler Application
 headerStreamHandler db = Tagged $ \req respond -> do
-    chan <- newChan
-    snd <$> concurrently
-        (SP.mapM_ (g >=> writeChan chan . f) . SP.concat $ blockDiffStream db)
-        (eventSourceAppChan chan req respond)
+    streamRef <- newIORef $ SP.map f $ SP.mapM g $ SP.concat $ blockDiffStream db
```

Comment:
Why not compose `f` and `g` within the `mapM`?

Reply:
Wouldn't that be the same? May understanding is that `SP.map f . SP.mapM g` get's fused into `SP.mapM (fmap f . g)`. But I find the former easier to read.